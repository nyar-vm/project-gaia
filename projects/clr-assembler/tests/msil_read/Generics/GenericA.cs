namespace Valkyrie.Translator.Generics;

public class GenericA<T>
{
    public T Value { get; set; }
}
