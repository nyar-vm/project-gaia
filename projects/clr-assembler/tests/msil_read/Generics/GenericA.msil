.class public auto ansi beforefieldinit
  Valkyrie.Translator.Generics.GenericA`1<T>
    extends [System.Runtime]System.Object
{
  .custom instance void [System.Runtime]System.Runtime.CompilerServices.NullableContextAttribute::.ctor(unsigned int8)
    = (01 00 01 00 00 ) // .....
    // unsigned int8(1) // 0x01
  .custom instance void [System.Runtime]System.Runtime.CompilerServices.NullableAttribute::.ctor(unsigned int8)
    = (01 00 00 00 00 ) // .....
    // unsigned int8(0) // 0x00
  .param type [1] /*T*/
    .custom instance void [System.Runtime]System.Runtime.CompilerServices.NullableAttribute::.ctor(unsigned int8)
      = (01 00 02 00 00 ) // .....
      // unsigned int8(2) // 0x02

  .field private !0/*T*/ '<Value>k__BackingField'
    .custom instance void [System.Runtime]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor()
      = (01 00 00 00 )
    .custom instance void [System.Runtime]System.Diagnostics.DebuggerBrowsableAttribute::.ctor(valuetype [System.Runtime]System.Diagnostics.DebuggerBrowsableState)
      = (01 00 00 00 00 00 00 00 ) // ........
      // int32(0) // 0x00000000

  .method public hidebysig specialname instance !0/*T*/
    get_Value() cil managed
  {
    .custom instance void [System.Runtime]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor()
      = (01 00 00 00 )
    .maxstack 8

    // [5 22 - 5 26]
    IL_0000: ldarg.0      // this
    IL_0001: ldfld        !0/*T*/ class Valkyrie.Translator.Generics.GenericA`1<!0/*T*/>::'<Value>k__BackingField'
    IL_0006: ret

  } // end of method GenericA`1::get_Value

  .method public hidebysig specialname instance void
    set_Value(
      !0/*T*/ 'value'
    ) cil managed
  {
    .custom instance void [System.Runtime]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor()
      = (01 00 00 00 )
    .maxstack 8

    // [5 27 - 5 31]
    IL_0000: ldarg.0      // this
    IL_0001: ldarg.1      // 'value'
    IL_0002: stfld        !0/*T*/ class Valkyrie.Translator.Generics.GenericA`1<!0/*T*/>::'<Value>k__BackingField'
    IL_0007: ret

  } // end of method GenericA`1::set_Value

  .method public hidebysig specialname rtspecialname instance void
    .ctor() cil managed
  {
    .maxstack 8

    IL_0000: ldarg.0      // this
    IL_0001: call         instance void [System.Runtime]System.Object::.ctor()
    IL_0006: nop
    IL_0007: ret

  } // end of method GenericA`1::.ctor

  .property instance !0/*T*/ Value()
  {
    .get instance !0/*T*/ Valkyrie.Translator.Generics.GenericA`1::get_Value()
    .set instance void Valkyrie.Translator.Generics.GenericA`1::set_Value(!0/*T*/)
  } // end of property GenericA`1::Value
} // end of class Valkyrie.Translator.Generics.GenericA`1
