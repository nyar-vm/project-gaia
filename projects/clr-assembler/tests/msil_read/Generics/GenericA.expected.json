{
    "assembly_name": null,
    "extern_assemblies": [],
    "module_name": null,
    "class_name": "Valkyrie.Translator.Generics.GenericA",
    "class_modifiers": [
        "public",
        "auto",
        "ansi",
        "beforefieldinit"
    ],
    "extends": null,
    "method_count": 0,
    "method_names": [],
    "statement_count": 1
}